### PR TITLE
[GSOC][TMVA][SOFIE] Fixed the implementation of MaxPool ONNX operator for 1d and 3d case 

### PR DESCRIPTION
**Pull request by Neel-Shah-29**
# This Pull request: Fixes the Implementation of Max Pool operator for 1d and 3d cases.
Earlier it was giving a runtime error for 1d and 3d cases of Max Pool operator. 
Error is described here.
image
I have also added the unit test for Max Pool 1d Operator.

## Checklist:

- [X] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

File: tmva/sofie/inc/TMVA/ROperator_Pool.hxx [hunk 1]
```diff
@@ -84,97 +85,152 @@ public:
...
+      // or it can be [N, C, F1,F2,....FN] . Minimum dimension is 3
       if (input.size() != 1 ) {
          throw std::runtime_error("TMVA SOFIE" + Name() + "Op Shape inference need 1 input tensor");
       }
       if (input[0].size() < 3) {
          throw std::runtime_error("TMVA SOFIE" + Name() + "Op Shape inference only accept tensor with at least 3 dimensions");
       }
-      // for the time being support only 4 dimensions
-      if (input[0].size() !=  4) {
+      // for the time being support only 3,4,5 dimens
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_Pool.hxx:102]: support only input tensors with dim=3,4,5

File: tmva/sofie/inc/TMVA/ROperator_Pool.hxx [hunk 2]
```diff
@@ -84,97 +85,152 @@ public:
...
+         if (fDim > 1)
+            fAttrKernelShape[1] = fShapeX[3];
+         if (fDim > 2)
+            fAttrKernelShape[2] = fShapeX[4];
          fAttrAutopad = "VALID";
-         fAttrPads = {0, 0, 0, 0};
+         fAttrPads = {0, 0, 0, 0, 0, 0 };
          assert(fAttrStrides.empty());
       }
-      // find shape of Y and add it in the list of intermediate tensors
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_Pool.hxx:170]: it should be intermediate

**Review by lmoneta — APPROVED** [review]
LGTM !
Thank you for the contribution, just some small correction on the code comments

File: tmva/sofie/inc/TMVA/ROperator_Pool.hxx (hunk 2 above)
Comment (Neel-Shah-29) [reply tmva/sofie/inc/TMVA/ROperator_Pool.hxx:170]: Done.

File: tmva/sofie/inc/TMVA/ROperator_Pool.hxx (hunk 1 above)
Comment (Neel-Shah-29) [reply tmva/sofie/inc/TMVA/ROperator_Pool.hxx]: Done.

**Pull request merged**